src/promnesia/sources/viber_desktop.py

```diff
@@ -0,0 +1,164 @@
+        ), G2 AS (
+            SELECT * FROM G0 WHERE
+                ContactID <> 1 AND
+                ChatId NOT IN (SELECT ChatId FROM G1)
+        ), Groups AS (
+            SELECT ChatId, chatname, PGTags FROM G1
+            UNION
+            SELECT ChatId, chatname, PGTags FROM G2
+        )
+        SELECT
+            M.EventId           AS mid,
+            E.TimeStamp         AS time,
+            G.chatname          AS chatname,
+            coalesce(
+                S.Name,
+                S.ClientName,
+                '(' || S.Number || ')'          /* contacts have one xor the other, but failsafe */
+            )                   AS sender,
+            coalesce(M.Subject, M.Body)         /* didn't see any msg with both */    
+                                AS text,
+            M.info              AS infojson,    /* to harvested titles from embedded urls */
+            G.PGTags            AS tags
+        FROM messages AS M
+        LEFT JOIN Events AS E
+            ON M.EventId = E.EventId
+        LEFT JOIN Contact AS S
+            ON E.ContactId = S.ContactId
+        LEFT JOIN Groups AS G
+            ON E.ChatId = G.ChatId
+        WHERE
+            text LIKE '%http%'
+        ORDER BY time;
+        """
+    )
+
+    def _parse_json_title(js) -> str:
+        if js and js.strip():
+            js = json.loads(js)
+            if isinstance(js, dict):
+                return js.get("Title")
+
+    def _handle_row(row) -> Results:
+        text = row["text"]
+        assert (
+            text
+        ), f"sql-query should have eliminated messages not containing 'http': {text}"
+        urls = extract_urls(text)
+        if not urls:
+            return
+        dt = from_epoch(row["time"] // 1000)  # timestamps are stored x100 this db
+        mid: str = unwrap(row["mid"])
+        # TODO perhaps we could be defensive with null sender/chat etc and still emit the Visit
+        sender: str = unwrap(row["sender"])
+        chatname: str = unwrap(row["chatname"])
+        sender: str = unwrap(row["sender"])
+        tags: str = unwrap(row["tags"])
+        infojson: str = unwrap(row["infojson"])
+
+        if tags and tags.strip():
+            tags = "".join(f"#{t}" for t in tags.split())
+            text = f"{text}\n\n{tags}"
+
+        url_title = _parse_json_title(infojson)
+        if url_title:
+            text = f"title: {url_title}\n\n{text}"
+
+        for u in urls:
+            # https://www.reddit.com/r/Telegram/comments/6ufwi3/link_to_a_specific_message_in_a_channel_possible/
+            # hmm, only seems to work on mobile app, but better than nothing...
+            yield Visit(
+                url=unquote(u),
+                dt=dt,
+                context=text,
+                locator=Loc.make(
+                    title=f"chat({mid}) from {sender}@{chatname}",
+                    href=f"sqlite://{database}?immutable=1#!Messages.EventId={mid}",
```

Comment:
interesting! does it actually work, i.e. if you press on the link it can jump to this message in sqlitebrowser? (or what are you using?)

Reply:
Ofcourse not :-)
 
But i needed something to contain internal information to help the use locate the message, besides the chat-name & the date, already visible in the sidebar.  And i wanted to be clear that it requires _programmatic-access_, hence the `sqlite` url-schema, which can be used verbatim as a connection URI.
I think he fragment would not matter, but haven't checked it.

Is there some other idea (like telegram forwarding- url service)?

Reply:
I managed to open the Viber DB in `sqlitebrowser` in linux, 
by ifirst making a "hard" link to the db-file, replacing the suffix from `.db --> .sqlite`,
an then simply referring to it with `file:///...sqlite` uri.

Would that be ok (for `promnesia index` to make a hard-link in huser's viber-config directory??

Reply:
I implemented the `file:://...sqlite` locator-URIs in df948c3.

Unfortunately, the links do not work, due to browser's security preventing opening local file-links, i guess,
because when i copy-past the link in another tab, it does offer me to open it with `sqlitebrowser`.

Reply:
> I think the fragment with the table.column=<mid> criterio would not matter

Yeah fair enough. Just asked in case it was some hidden feature, it could be convenient :) 

I think the closest possible thing that sqlitebrowser has is `--sql` argument (although I couldn't get it to work), but it wouldn't work file `file://` handler unfortunately.

> Is there some other idea (like telegram forwarding- url service)?

Quick search gives something: https://stackoverflow.com/a/41324451/706389, but it's probably unlikely to work on desktop unfortunately :( Such things are always better on mobile.

> Would that be ok (for promnesia index to make a hard-link in huser's viber-config directory??

Hmm. Hardlinks are pretty extreme and opaque IMO -- the user might damage the original database without knowing it. But I see the problem with symlinks, they seem to be resolved before opening and it then tries to open '.db' file.
Maybe comment the hardlink logic for now? I'm worried that someone trashes their database; maybe we can come up with a more robust idea later? 
Also see the comment https://github.com/karlicoss/promnesia/pull/204#issuecomment-785374009 how to handle it on user site without hardlinking (a little tedious, but you only need to do it once, and maybe later we can integrate these instructions in promnesia, similar to [open-with-editor(https://github.com/karlicoss/open-in-editor) ?)

Reply:
Great research.
Will drop the hard-link part.  Closing this review.